### ISSUE TITLE
CleanURLs module process percent-encoding query params incorrectly

### ISSUE THREAD

**Issue by JasonHK** [1fexd/LinkSheet#311]
### Bug description

Sample URLs:
- [https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=**%E5%8F%B0%E5%A0%B4**&padt=117&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1&ie=UTF-8](https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=%E5%8F%B0%E5%A0%B4&padt=117&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1&ie=UTF-8)
- [https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=**%E6%9D%B1%E7%AB%8B%E5%B0%8F%E8%AA%AA%E9%A0%90%E5%AE%9A%E5%87%BA%E6%9B%B8**&showCalendars=0&mode=agenda](https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=%E6%9D%B1%E7%AB%8B%E5%B0%8F%E8%AA%AA%E9%A0%90%E5%AE%9A%E5%87%BA%E6%9B%B8&showCalendars=0&mode=agenda)

The part highlighted in bold got replaced to "�4" and "q�����" respectively, for some reason.

### Version

0.0.33

### Device

Samsung Galaxy S21

### Android version

Android 14 (OneUI 6.0)

### Application log or crash log

```
1707035720823 D IntentResolver: Intent=act=android.intent.action.VIEW,cat={android.intent.category.BROWSABLE},cmp=ComponentInfo{fe.linksheet/fe.linksheet.activity.BottomSheetActivity},flags=276824064
1707035720823 D IntentExt: GetUri: Pre modification=https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=%E6%9D%B1%E7%AB%8B%E5%B0%8F%E8%AA%AA%E9%A0%90%E5%AE%9A%E5%87%BA%E6%9B%B8&showCalendars=0&mode=agenda
1707035720831 D IntentExt: GetUri: Post FastForward=https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=%E6%9D%B1%E7%AB%8B%E5%B0%8F%E8%AA%AA%E9%A0%90%E5%AE%9A%E5%87%BA%E6%9B%B8&showCalendars=0&mode=agenda
1707035720832 D IntentExt: GetUri: Post ClearURL=https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=%E6%9D%B1%E7%AB%8B%E5%B0%8F%E8%AA%AA%E9%A0%90%E5%AE%9A%E5%87%BA%E6%9B%B8&showCalendars=0&mode=agenda
1707035720893 D RedirectUrlResolver: Following redirects for https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=%E6%9D%B1%E7%AB%8B%E5%B0%8F%E8%AA%AA%E9%A0%90%E5%AE%9A%E5%87%BA%E6%9B%B8&showCalendars=0&mode=agenda
1707035720985 D IntentResolver: PreferredApp=preferredApp={host=calendar.google.com,pkg=org.mozilla.fenix,cmp=ComponentInfo{org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity},alwaysPreferred=false},displayActivityInfo={activityInfo=ActivityInfo{ff51c27 org.mozilla.fenix.App},label=Firefox Nightly,resolveInfo=ResolveInfo{71b0ed4 org.mozilla.fenix/.App m=0x108000 userHandle=UserHandle{0}}}
1707035720995 D IntentResolver: LastUsedApps={packageName=org.mozilla.fenix,lastUsed=1707034547738}
1707035720995 D IntentResolver: NewIntent=act=android.intent.action.VIEW,cat={android.intent.category.BROWSABLE},flags=268435456
1707035721028 D IntentResolver: ResolveList=[ResolveInfo{98d53c3 com.android.chrome/com.google.android.apps.chrome.IntentDispatcher m=0x208000 userHandle=UserHandle{0}},ResolveInfo{c844440 com.instantbits.cast.webvideo/com.instantbits.cast.BrowserComponent m=0x208000 userHandle=UserHandle{0}},ResolveInfo{a9cbf79 com.mxtech.videoplayer.pro/com.mxtech.videoplayer.ActivityWebBrowser m=0x208000 userHandle=UserHandle{0}},ResolveInfo{7aa94be com.sec.android.app.sbrowser/.SBrowserLauncherActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{5e7691f io.metamask/.MainActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{289646c me.webalert/.activity.CreateJobFromUrlActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{8a0da35 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{8f73eca org.mozilla.fennec_fdroid/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{a3783b org.mozilla.firefox_beta/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{cde5b58 org.torproject.torbrowser/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{891e4b1 jp.ddo.hotmist.unicodepad/.UnicodeActivity m=0x208000 userHandle=UserHandle{0}}]
1707035721032 D IntentResolver: BrowserMode=mode=browser,resolveInfo=ResolveInfo{bb27d96 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}}
1707035721050 D IntentResolver: Grouped=[], filteredItem=activityInfo=ActivityInfo{f10dd22 org.mozilla.fenix.IntentReceiverActivity},label=Firefox Nightly,resolveInfo=ResolveInfo{8a0da35 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}}, showExtended=false, selectedBrowserIsSingleOption=true, noBrowsersPresentOnlySingleApp=false
1707035721051 D BottomSheetViewModel: Component=ComponentInfo{org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity}
1707035721051 D BottomSheetViewModel: Inserting=host=calendar.google.com,pkg=org.mozilla.fenix,cmp=ComponentInfo{org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity},alwaysPreferred=false
1707035721053 D BottomSheetViewModel: Inserting={host=calendar.google.com,packageName=org.mozilla.fenix,lastUsed=1707035721052}
1707035743385 D IntentResolver: Intent=act=android.intent.action.VIEW,cat={android.intent.category.BROWSABLE},cmp=ComponentInfo{fe.linksheet/fe.linksheet.activity.BottomSheetActivity},flags=276824064
1707035743386 D IntentExt: GetUri: Pre modification=https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=%E5%8F%B0%E5%A0%B4&padt=117&gs_lp=Eg1nc2EtZ29vZ2xlYXBwGgIYACICIyMyCQgAEAAYgAQoATIJCAEQABiABCgBMgkIAhAAGIAEKAEyCQgDEAAYgAQoATIbCAQQLhivARjHARiABCgBagovbS8wbmJodGxkMgkIBRAAGIAEKAEyCQgGEAAYgAQoATIeCAcQLhivARjHARiABCgBag0vZy8xMWI3NmNnNF9qMhoICBAuGK8BGMcBGIAEKAFqCS9tLzBiejZxNTIJCAkQABiABCgBQAJI-V1Q7FhY7FhgAogBAZgBwAGgAcABqgEDMGoxsAEAuAEDmAICoALZAagCHrACAcICDxAuGCcYrwEYxwEY6gIoAcICCRAjGCcY6gIoAcICDBAjGCcY6gIY2AMoAcICCRAuGCcY6gIoAdICCy1tS2VZNEE2UHJNmAMRoAOKArAEALgEAMIEANAEAIIFBggAEAAoAPEFXPJBz9iP1RaABgI&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1&ie=UTF-8
1707035743404 D IntentExt: GetUri: Post FastForward=https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=%E5%8F%B0%E5%A0%B4&padt=117&gs_lp=Eg1nc2EtZ29vZ2xlYXBwGgIYACICIyMyCQgAEAAYgAQoATIJCAEQABiABCgBMgkIAhAAGIAEKAEyCQgDEAAYgAQoATIbCAQQLhivARjHARiABCgBagovbS8wbmJodGxkMgkIBRAAGIAEKAEyCQgGEAAYgAQoATIeCAcQLhivARjHARiABCgBag0vZy8xMWI3NmNnNF9qMhoICBAuGK8BGMcBGIAEKAFqCS9tLzBiejZxNTIJCAkQABiABCgBQAJI-V1Q7FhY7FhgAogBAZgBwAGgAcABqgEDMGoxsAEAuAEDmAICoALZAagCHrACAcICDxAuGCcYrwEYxwEY6gIoAcICCRAjGCcY6gIoAcICDBAjGCcY6gIY2AMoAcICCRAuGCcY6gIoAdICCy1tS2VZNEE2UHJNmAMRoAOKArAEALgEAMIEANAEAIIFBggAEAAoAPEFXPJBz9iP1RaABgI&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1&ie=UTF-8
1707035743404 D IntentExt: GetUri: Post ClearURL=https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=%E5%8F%B0%E5%A0%B4&padt=117&gs_lp=Eg1nc2EtZ29vZ2xlYXBwGgIYACICIyMyCQgAEAAYgAQoATIJCAEQABiABCgBMgkIAhAAGIAEKAEyCQgDEAAYgAQoATIbCAQQLhivARjHARiABCgBagovbS8wbmJodGxkMgkIBRAAGIAEKAEyCQgGEAAYgAQoATIeCAcQLhivARjHARiABCgBag0vZy8xMWI3NmNnNF9qMhoICBAuGK8BGMcBGIAEKAFqCS9tLzBiejZxNTIJCAkQABiABCgBQAJI-V1Q7FhY7FhgAogBAZgBwAGgAcABqgEDMGoxsAEAuAEDmAICoALZAagCHrACAcICDxAuGCcYrwEYxwEY6gIoAcICCRAjGCcY6gIoAcICDBAjGCcY6gIY2AMoAcICCRAuGCcY6gIoAdICCy1tS2VZNEE2UHJNmAMRoAOKArAEALgEAMIEANAEAIIFBggAEAAoAPEFXPJBz9iP1RaABgI&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1&ie=UTF-8
1707035743411 D RedirectUrlResolver: Following redirects for https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=%E5%8F%B0%E5%A0%B4&padt=117&gs_lp=Eg1nc2EtZ29vZ2xlYXBwGgIYACICIyMyCQgAEAAYgAQoATIJCAEQABiABCgBMgkIAhAAGIAEKAEyCQgDEAAYgAQoATIbCAQQLhivARjHARiABCgBagovbS8wbmJodGxkMgkIBRAAGIAEKAEyCQgGEAAYgAQoATIeCAcQLhivARjHARiABCgBag0vZy8xMWI3NmNnNF9qMhoICBAuGK8BGMcBGIAEKAFqCS9tLzBiejZxNTIJCAkQABiABCgBQAJI-V1Q7FhY7FhgAogBAZgBwAGgAcABqgEDMGoxsAEAuAEDmAICoALZAagCHrACAcICDxAuGCcYrwEYxwEY6gIoAcICCRAjGCcY6gIoAcICDBAjGCcY6gIY2AMoAcICCRAuGCcY6gIoAdICCy1tS2VZNEE2UHJNmAMRoAOKArAEALgEAMIEANAEAIIFBggAEAAoAPEFXPJBz9iP1RaABgI&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1&ie=UTF-8
1707035743500 D IntentResolver: PreferredApp=preferredApp={host=maps.google.com,pkg=com.google.android.apps.maps,cmp=ComponentInfo{com.google.android.apps.maps/com.google.android.maps.MapsActivity},alwaysPreferred=false},displayActivityInfo={activityInfo=ActivityInfo{4b50135 com.google.android.maps.MapsActivity},label=地圖,resolveInfo=ResolveInfo{b6dd9ca com.google.android.apps.maps/com.google.android.maps.MapsActivity m=0x108000 userHandle=UserHandle{0}}}
1707035743512 D IntentResolver: LastUsedApps={packageName=com.google.android.apps.maps,lastUsed=1706682250432}
1707035743512 D IntentResolver: NewIntent=act=android.intent.action.VIEW,cat={android.intent.category.BROWSABLE},flags=268435456
1707035743525 D IntentResolver: ResolveList=[ResolveInfo{ec1ee58 com.google.android.apps.maps/com.google.android.maps.MapsActivity m=0x508000 userHandle=UserHandle{0}},ResolveInfo{f417bb1 net.osmand.plus/.activities.search.GeoIntentActivity m=0x308000 userHandle=UserHandle{0}},ResolveInfo{e77c896 com.android.chrome/com.google.android.apps.chrome.IntentDispatcher m=0x208000 userHandle=UserHandle{0}},ResolveInfo{2224c17 com.instantbits.cast.webvideo/com.instantbits.cast.BrowserComponent m=0x208000 userHandle=UserHandle{0}},ResolveInfo{c939804 com.mxtech.videoplayer.pro/com.mxtech.videoplayer.ActivityWebBrowser m=0x208000 userHandle=UserHandle{0}},ResolveInfo{a4121ed com.sec.android.app.sbrowser/.SBrowserLauncherActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{5fd822 io.metamask/.MainActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{8f592b3 me.webalert/.activity.CreateJobFromUrlActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{cb43070 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{61aefe9 org.mozilla.fennec_fdroid/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{5c7546e org.mozilla.firefox_beta/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{caca70f org.torproject.torbrowser/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{503e39c jp.ddo.hotmist.unicodepad/.UnicodeActivity m=0x208000 userHandle=UserHandle{0}}]
1707035743528 D IntentResolver: BrowserMode=mode=browser,resolveInfo=ResolveInfo{8a9a1a5 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}}
1707035743710 D IntentResolver: Grouped=[activityInfo=ActivityInfo{5f9a1ce org.mozilla.fenix.IntentReceiverActivity},label=Firefox Nightly,resolveInfo=ResolveInfo{cb43070 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},activityInfo=ActivityInfo{dead8ef net.osmand.plus.activities.search.GeoIntentActivity},label=OsmAnd+,resolveInfo=ResolveInfo{f417bb1 net.osmand.plus/.activities.search.GeoIntentActivity m=0x308000 userHandle=UserHandle{0}}], filteredItem=activityInfo=ActivityInfo{2f15ffc com.google.android.maps.MapsActivity},label=地圖,resolveInfo=ResolveInfo{ec1ee58 com.google.android.apps.maps/com.google.android.maps.MapsActivity m=0x508000 userHandle=UserHandle{0}}, showExtended=false, selectedBrowserIsSingleOption=false, noBrowsersPresentOnlySingleApp=false
1707035744545 D BottomSheetViewModel: Component=ComponentInfo{com.google.android.apps.maps/com.google.android.maps.MapsActivity}
1707035744545 D BottomSheetViewModel: Inserting=host=maps.google.com,pkg=com.google.android.apps.maps,cmp=ComponentInfo{com.google.android.apps.maps/com.google.android.maps.MapsActivity},alwaysPreferred=false
1707035744551 D BottomSheetViewModel: Inserting={host=maps.google.com,packageName=com.google.android.apps.maps,lastUsed=1707035744548}
1707035758384 D IntentResolver: Intent=act=android.intent.action.VIEW,cat={android.intent.category.BROWSABLE},cmp=ComponentInfo{fe.linksheet/fe.linksheet.activity.BottomSheetActivity},flags=276824064
1707035758384 D IntentExt: GetUri: Pre modification=https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=%E5%8F%B0%E5%A0%B4&padt=117&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1&ie=UTF-8
1707035758386 D IntentExt: GetUri: Post FastForward=https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=%E5%8F%B0%E5%A0%B4&padt=117&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1&ie=UTF-8
1707035758429 D IntentExt: GetUri: Post ClearURL=https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=�4&padt=117&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1
1707035758535 D RedirectUrlResolver: Following redirects for https://maps.google.com/maps?hl=zh-HK&ram_mb=7362&aos=10&vpa=1&cct_optout=1&ampcct=6167&lns_as=1&cs=0&q=�4&padt=117&rdid=8e79aa71-d72b-4d62-9d0b-49a8abfbdff2&client=ms-android-samsung-rvo1&um=1
1707035758568 D IntentResolver: PreferredApp=preferredApp={host=maps.google.com,pkg=com.google.android.apps.maps,cmp=ComponentInfo{com.google.android.apps.maps/com.google.android.maps.MapsActivity},alwaysPreferred=false},displayActivityInfo={activityInfo=ActivityInfo{8f98416 com.google.android.maps.MapsActivity},label=地圖,resolveInfo=ResolveInfo{7802197 com.google.android.apps.maps/com.google.android.maps.MapsActivity m=0x108000 userHandle=UserHandle{0}}}
1707035758601 D IntentResolver: LastUsedApps={packageName=com.google.android.apps.maps,lastUsed=1707035744548}
1707035758601 D IntentResolver: NewIntent=act=android.intent.action.VIEW,cat={android.intent.category.BROWSABLE},flags=268435456
1707035758611 D IntentResolver: ResolveList=[ResolveInfo{401cba2 com.google.android.apps.maps/com.google.android.maps.MapsActivity m=0x508000 userHandle=UserHandle{0}},ResolveInfo{21b4033 net.osmand.plus/.activities.search.GeoIntentActivity m=0x308000 userHandle=UserHandle{0}},ResolveInfo{d9c0ff0 com.android.chrome/com.google.android.apps.chrome.IntentDispatcher m=0x208000 userHandle=UserHandle{0}},ResolveInfo{bee1969 com.instantbits.cast.webvideo/com.instantbits.cast.BrowserComponent m=0x208000 userHandle=UserHandle{0}},ResolveInfo{54ffee com.mxtech.videoplayer.pro/com.mxtech.videoplayer.ActivityWebBrowser m=0x208000 userHandle=UserHandle{0}},ResolveInfo{a3fac8f com.sec.android.app.sbrowser/.SBrowserLauncherActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{94a3b1c io.metamask/.MainActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{645e325 me.webalert/.activity.CreateJobFromUrlActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{a952cfa org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{89482ab org.mozilla.fennec_fdroid/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{24ded08 org.mozilla.firefox_beta/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{5968ca1 org.torproject.torbrowser/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{5791ec6 jp.ddo.hotmist.unicodepad/.UnicodeActivity m=0x208000 userHandle=UserHandle{0}}]
1707035758631 D IntentResolver: BrowserMode=mode=browser,resolveInfo=ResolveInfo{f169e87 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}}
1707035758784 D IntentResolver: Grouped=[activityInfo=ActivityInfo{3648f68 org.mozilla.fenix.IntentReceiverActivity},label=Firefox Nightly,resolveInfo=ResolveInfo{a952cfa org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},activityInfo=ActivityInfo{7f2ef81 net.osmand.plus.activities.search.GeoIntentActivity},label=OsmAnd+,resolveInfo=ResolveInfo{21b4033 net.osmand.plus/.activities.search.GeoIntentActivity m=0x308000 userHandle=UserHandle{0}}], filteredItem=activityInfo=ActivityInfo{62f7826 com.google.android.maps.MapsActivity},label=地圖,resolveInfo=ResolveInfo{401cba2 com.google.android.apps.maps/com.google.android.maps.MapsActivity m=0x508000 userHandle=UserHandle{0}}, showExtended=false, selectedBrowserIsSingleOption=false, noBrowsersPresentOnlySingleApp=false
1707035759595 D BottomSheetViewModel: Component=ComponentInfo{com.google.android.apps.maps/com.google.android.maps.MapsActivity}
1707035759597 D BottomSheetViewModel: Inserting=host=maps.google.com,pkg=com.google.android.apps.maps,cmp=ComponentInfo{com.google.android.apps.maps/com.google.android.maps.MapsActivity},alwaysPreferred=false
1707035759599 D BottomSheetViewModel: Inserting={host=maps.google.com,packageName=com.google.android.apps.maps,lastUsed=1707035759598}
1707036088249 D IntentResolver: Intent=act=android.intent.action.VIEW,cat={android.intent.category.BROWSABLE},cmp=ComponentInfo{fe.linksheet/fe.linksheet.activity.BottomSheetActivity},flags=276824064
1707036088249 D IntentExt: GetUri: Pre modification=https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=%E6%9D%B1%E7%AB%8B%E5%B0%8F%E8%AA%AA%E9%A0%90%E5%AE%9A%E5%87%BA%E6%9B%B8&showCalendars=0&mode=agenda
1707036088251 D IntentExt: GetUri: Post FastForward=https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=%E6%9D%B1%E7%AB%8B%E5%B0%8F%E8%AA%AA%E9%A0%90%E5%AE%9A%E5%87%BA%E6%9B%B8&showCalendars=0&mode=agenda
1707036088258 D IntentExt: GetUri: Post ClearURL=https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=q�����&showCalendars=0&mode=agenda
1707036088273 D RedirectUrlResolver: Following redirects for https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=q�����&showCalendars=0&mode=agenda
1707036088278 D Amp2HtmlUrlResolver: Following redirects for https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=q�����&showCalendars=0&mode=agenda
1707036088279 D Amp2HtmlUrlResolver: Using local service for https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=q�����&showCalendars=0&mode=agenda
1707036088281 D Amp2HtmlResolveRequest: ResolveLocal https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=q�����&showCalendars=0&mode=agenda
1707036088282 D CachedRequest: No cached response found for GET https://calendar.google.com/calendar/embed?src=tongli.novels@gmail.com&title=q�����&showCalendars=0&mode=agenda, sending request..
1707036089113 D IntentResolver: PreferredApp=preferredApp={host=calendar.google.com,pkg=org.mozilla.fenix,cmp=ComponentInfo{org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity},alwaysPreferred=false},displayActivityInfo={activityInfo=ActivityInfo{9c2f8be org.mozilla.fenix.App},label=Firefox Nightly,resolveInfo=ResolveInfo{c1fd1f org.mozilla.fenix/.App m=0x108000 userHandle=UserHandle{0}}}
1707036089116 D IntentResolver: LastUsedApps={packageName=org.mozilla.fenix,lastUsed=1707035721052}
1707036089116 D IntentResolver: NewIntent=act=android.intent.action.VIEW,cat={android.intent.category.BROWSABLE},flags=268435456
1707036089122 D IntentResolver: ResolveList=[ResolveInfo{14f8e35 com.android.chrome/com.google.android.apps.chrome.IntentDispatcher m=0x208000 userHandle=UserHandle{0}},ResolveInfo{e31e2ca com.instantbits.cast.webvideo/com.instantbits.cast.BrowserComponent m=0x208000 userHandle=UserHandle{0}},ResolveInfo{934c3b com.mxtech.videoplayer.pro/com.mxtech.videoplayer.ActivityWebBrowser m=0x208000 userHandle=UserHandle{0}},ResolveInfo{939f58 com.sec.android.app.sbrowser/.SBrowserLauncherActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{f77d8b1 io.metamask/.MainActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{3f36196 me.webalert/.activity.CreateJobFromUrlActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{6607117 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{84f5904 org.mozilla.fennec_fdroid/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{4004eed org.mozilla.firefox_beta/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{47c0122 org.torproject.torbrowser/org.mozilla.fenix.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}},ResolveInfo{37107b3 jp.ddo.hotmist.unicodepad/.UnicodeActivity m=0x208000 userHandle=UserHandle{0}}]
1707036089124 D IntentResolver: BrowserMode=mode=browser,resolveInfo=ResolveInfo{eeb0170 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}}
1707036089138 D IntentResolver: Grouped=[], filteredItem=activityInfo=ActivityInfo{256c49c org.mozilla.fenix.IntentReceiverActivity},label=Firefox Nightly,resolveInfo=ResolveInfo{6607117 org.mozilla.fenix/.IntentReceiverActivity m=0x208000 userHandle=UserHandle{0}}, showExtended=false, selectedBrowserIsSingleOption=true, noBrowsersPresentOnlySingleApp=false
1707036089139 D BottomSheetViewModel: Component=ComponentInfo{org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity}
1707036089139 D BottomSheetViewModel: Inserting=host=calendar.google.com,pkg=org.mozilla.fenix,cmp=ComponentInfo{org.mozilla.fenix/org.mozilla.fenix.IntentReceiverActivity},alwaysPreferred=false
1707036089143 D BottomSheetViewModel: Inserting={host=calendar.google.com,packageName=org.mozilla.fenix,lastUsed=1707036089141}

linksheet_version=0.0.33,code=33
device_fingerprint=samsung/o1qzhx/o1q:14/UP1A.231005.007/G9910ZHU6FWK7:user/release-keys
preferences:
enable_copy_button=false
hide_after_copying=false
single_tap=true
usage_stats_sorting=true
browser_mode=null
in_app_browser_mode=null
unified_preferred_browser=true
in_app_browser_setting=null
enable_send_button=true
always_show_package_name=false
disable_toasts=true
grid_layout=false
use_clear_urls=true
fast_forward_rules=true
enable_lib_redirect=false
follow_redirects=true
follow_redirects=true
follow_redirects_local_cache=true
follow_redirects_external_service=false
follow_only_known_trackers=true
theme=null
dont_show_filtered_item=false
use_text_share_copy_buttons=false
preview_url=false
enable_downloader=false
downloaderCheckUrlMimeType=false
enable_ignore_lib_redirect_button=false
follow_redirects_timeout=15
enable_request_private_browsing_button=false
enable_amp2html=true
amp2html_local_cache=true
amp2html_external_service=false
follow_redirects_builtin_cache=true
amp2html_builtin_cache=true
selected_browserorg.mozilla.fenix
selected_in_app_browsernull
```

### Reproducible on latest nightly build

Yes

**Comment by 1fexd** [comment]
`0.0.33` is very old, please check if the issue is also present in the latest nightly build.

**Comment by JasonHK** [comment]
Yes the issue still present on the latest build c9fe609

**Comment by SodaWithoutSparkles** [comment]
I've got a more simple URL to test: https://www.google.com/search?q=%22Linksheet%22

I believe basically anything non-ascii is affected

and i want to say that @JasonHK had a lot of info leaked in the URL, including your email and other potentially important information. You can edit them out, and remove the history by following the steps in [here](https://stackoverflow.com/questions/69755177/how-to-remove-sensitive-information-from-issue-history-on-github)

**Comment by JasonHK** [comment]
@SodaWithoutSparkles Thanks for your reminder, I've updated the comment and removed the revision for now. 

P.S. That email address belongs to a book publisher which they used to provide a public Google calendar on their website for book publishing schedule.

**Comment by SodaWithoutSparkles** [comment]
it seems that the mysterious symbol is the [unicode symbol \u{FFFD}](https://r12a.github.io/app-conversion/). The unicode symbol is a "Replacement Character" according to [this site](https://www.compart.com/en/unicode/U+FFFD) and the [unicode database](ftp://ftp.unicode.org/Public/UNIDATA/UnicodeData.txt).

Sidenote: @JasonHK I found a few issues regarding your site, as i don't think it would be approiate to dicuss that here, you can contact me via the email (alias) in my bio.

**Comment by SodaWithoutSparkles** [comment]
I think my last assumption was incorrect. Maybe all CJK-encoded URLs would be parsed incorrectly

- C: https://www.google.com/search?q=%E4%B8%AD%E6%96%87
- J: https://www.google.com/search?q=%E3%81%95%E3%82%88%E3%81%86%E3%81%AA%E3%82%89
- K: Not yet tested, will update later

**Comment by SodaWithoutSparkles** [comment]
After some more investigation, it turns out the "open from clipboard" feature just launches the default browser. Is this intended?

I did not set the nightly build as default browser last time I tried, therefore the old linksheet was launched. Can confirm the fix was indeed working now.

**Comment by 1fexd** [comment]
> After some more investigation, it turns out the "open from clipboard" feature just launches the default browser. Is this intended?
> 
> I did not set the nightly build as default browser last time I tried, therefore the old linksheet was launched. Can confirm the fix was indeed working now.

Is this related to this issue?

**Comment by SodaWithoutSparkles** [comment]
> When are you experiencing this? I can't reproduce it on the latest nightly build.

TL;DR: This issue is resolved and could be closed.

The issue is that, when I was testing, I didn't fully setup the nightly build (refered as *nightly*) as default browser. I copied the URL and opened *nightly*, then used the "Open from clipboard" function. It turns out that the function just launches the default browser (which was still set to 0.0.34), therefore it won't work (obviously). This prompted me to raise the clipboard comment.

Then I proceed to fully setup *nightly* (hence issue 324) and tried to reproduce this issue again, but this time the issue is resolved. 

I'll open another issue regarding the unexpected behaviour of "open from clipboard" in both stable and nightly builds.
